lnrpc/routerrpc/router.proto

```diff
@@ -334,6 +343,14 @@ message QueryMissionControlResponse {
     repeated PairHistory pairs = 2;
 }
 
+message ImportMissionControlRequest {
+    // Node pair-level mission control state to be imported.
+    repeated PairHistory pairs = 1;
```

Comment:
Mission control's internal state on disk isn't the pair history, but the low level raw payment attempt data with routes and everything. The reason for storing that is that it would allow the mission control logic to be modified without the need for data migration (if possible at all with the derived data).

Another benefit of operating on raw data is that it is possible to change mission control settings and re-derive its state again. In the future, this could even be done runtime. Actually it could be part of the other mission control PR that allows changing the settings runtime. 

But because everything is internal, we could always find a way to deal with an upgrade.

With this PR however, that becomes more troublesome. This RPC freezes the structure of mission control's internal state. Changing it will involve a breaking change or supporting the older RPC. To me that doesn't look ideal.

The alternative is to create an import feature for raw payment data. This import function would reconstruct mission control's state in whatever format is best. Sensitive fields like the payment hash could be stripped out as long as they aren't processed to reconstruct mission control's state.

Reply:
> Mission control's internal state on disk isn't the pair history, but the low level raw payment attempt data with routes and everything

The internal state _is_ this currently, but we do indeed derive that data from the raw payment routes. We went in this direction as we're looking at a basic form of distributed mission control where nodes can start to share this information. By updating only the resulting _in-memory_state, we don't need to also store all the route information which is larger, and leaks more information w.r.t the activity of a given node. Also note that this is intended to just be in memory and not propagate to what's stored on disk yet. We could even consider just create a parallel ephemeral mission control where this new data is applied. 

> With this PR however, that becomes more troublesome. This RPC freezes the structure of mission control's internal state.

Eh not really, it's a new experimental call that we're open to breaking at will for the purpose of our experiments. The other advantage of doing something like this in the short term is that the serialized information is _much_ smaller. A hundred individual payment attempts utilizing a total of 1k individual routes can be compressed into a single pair (not likely in reality, just an example). 

> The alternative is to create an import feature for raw payment data. This import function would reconstruct mission control's state in whatever format is best

We considered this but dismissed it in the short term due to the drawbacks I mentioned above. If you want to take a stab at implementing a stripped down version, we'd consider it, but as pointed out above it would leak _a lot_ more information (the payment hash is the least of our concerns).

Reply:
We can also leave a path open to the raw data import by making this field a `oneof` w/ the current implementation being a simple in-memory insertion. In the future, if we choose to go with the raw scrubbed data format, we can deprecate the pair history over time. Thoughts @carlaKC ?

Reply:
>The internal state is this currently

I wrote 'Mission control's internal state *on disk*...' I think my statement is correct.

>Eh not really, it's a new experimental call that we're open to breaking at will for the purpose of our experiments.

That wasn't obvious to me from looking at the change set. It is in the router subserver which is now enabled by default iirc. 

>The other advantage of doing something like this in the short term is that the serialized information is much smaller

Question is whether this advantage is relevant in practice for users.

I get the downside of leaking info with raw payment history. But isn't 'distributed mission control' intended to be used by a node operator that runs multiple nodes? In that case nothing is leaked that the operator didn't already know.

If the idea is to dump these exports on the internet, I can understand that some won't do that because of the sensitivity of the information. But it isn't said that the current in-memory state can't be used to derive interesting information from as well. It isn't as rich as the raw data, but could still present a privacy risk.

Reply:
To me there are two different kinds of imports that one could want for MC:
1. Public import: obtain information from an external source to boost an inactive/new node, privacy concerns for data exporter
2. Personal import: share information between your own nodes, to bootstrap a new node coming online or restore from a previously active node, no privacy concerns

This PR was intended to solve (1), my fault that it wasn't very clear, since newer nodes struggling to route has been a more reported problem than the need to bootstrap your own nodes (2). If we get some feature requests for the latter, I think it would be a great thing to add as well. 

>  But it isn't said that the current in-memory state can't be used to derive interesting information from as well. It isn't as rich as the raw data, but could still present a privacy risk.

This is indeed an open question, but I do think that the aggregation of results does obfuscate routing information to a large degree, especially since one would expect larger nodes to be providing this export. This information is also optionally exported and shared at the discretion of the owner - will add a warning note to the export proto, and be sure to mention this in release notes as well.   

> leave a path open to the raw data import by making this field a `oneof`

As for a one-of in the RPC, I think that we can achieve the same with flat fields in the proto then just logically enforcing that only one of them is set? I've been a bit put off by one-of recently, but that's personal preference so happy to update it.

Reply:
Great points @carlaKC, yeah we're only after #1 here, but that doesn't block off a future path doing #2 later. @joostjager  wasn't keyed into the discussions we'd had w.r.t our end goals internally, but hopefully that's now clear. 

Correct the current pair-wise data can leak information, but it's demonstrably less than the set of paths in their totality. 

> As for a one-of in the RPC, I think that we can achieve the same with flat fields in the proto then just logically enforcing that only one of them is set? I've been a bit put off by one-of recently, but that's personal preference so happy to update it.

Up to you really, no strong preference. FWIW, the latest versions of protos have made working w/ oneofs a bit easier with added helper methods. Though the naming of the fields are still rather ugly, at least when manipulating them in Go.

Reply:
Yes, it is clear now. I was on the track of import type 2, 'distributed mission control'. For more public datasets it makes sense to use an aggregated format to obfuscate sensitive data.